hkmc2/shared/src/main/scala/hkmc2/codegen/Lowering.scala

```diff
@@ -551,17 +557,34 @@ class Lowering()(using Config, TL, Raise, State, Ctx):
           k(DynSelect(p, f, ai))
       
       
-    case New(cls, as, N) =>
+    case New(cls, ass, N) =>
       subTerm(cls): sr =>
-        subTerms(as): asr =>
+        val head = ass.headOr(Nil)
+        val tail = ass.tailOr(Nil)
+        tail match
+        case Nil => subTerms(head): asr =>
           k(Instantiate(sr, asr))
+        case tail => subTerms(head): asr =>
+          val z = tail.foldLeft[Path => Block](k): (acc, args) => 
+            inner =>
+              plainArgs(args): args =>
+                val ts = TempSymbol(N)
+                Assign(ts, Call(inner, args)(true, true), acc(Value.Ref(ts)))
+          val ts = TempSymbol(N)
+          Assign(ts, Instantiate(sr, asr), z(Value.Ref(ts)))
       
-    case New(cls, as, S((isym, rft))) =>
+    case New(cls, ass, S((isym, rft))) =>
       subTerm(cls): clsp =>
         val sym = new BlockMemberSymbol(isym.name, Nil)
         val (mtds, publicFlds, privateFlds, ctor) = gatherMembers(rft)
+        if ass.length > 1 then 
```

Comment:
Why is this error logic duplicated witht the one above?

Reply:
Oh because the two pieces of lowering logic are just identical in the beginning. There was a comment `TODO dedup with New case` later

```
            val pctor = // TODO dedup with New case
              plainArgs(ext.argss.headOr(Nil)): args =>
                Return(Call(Value.Ref(State.builtinOpsMap("super")), args)(true, true), implct = true)
```